// @progress/kendo-react-layout/dist/npm/card/Avatar.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Avatar = void 0;
var React = require("react");
var PropTypes = require("prop-types");
var kendo_react_common_1 = require("@progress/kendo-react-common");
var Enums_1 = require("./interfaces/Enums");
var kendo_react_common_2 = require("@progress/kendo-react-common");
var package_metadata_1 = require("../package-metadata");
var Avatar = function (props) {
    var _a;
    (0, kendo_react_common_2.validatePackage)(package_metadata_1.packageMetadata);
    return (React.createElement("div", { style: props.style, className: (0, kendo_react_common_1.classNames)('k-avatar', (_a = {},
            _a['k-avatar-bordered'] = props.border,
            _a["k-rounded-".concat(kendo_react_common_1.kendoThemeMaps.roundedMap[props.rounded] || props.rounded)] = props.rounded,
            _a["k-avatar-".concat(kendo_react_common_1.kendoThemeMaps.sizeMap[props.size] || props.size)] = props.size,
            _a["k-avatar-".concat(props.fillMode)] = props.fillMode,
            _a["k-avatar-".concat(props.fillMode, "-").concat(props.themeColor)] = Boolean(props.fillMode && props.themeColor),
            _a), props.className) },
        React.createElement("span", { className: "k-avatar-".concat(props.type) }, props.children)));
};
exports.Avatar = Avatar;
exports.Avatar.propTypes = {
    className: PropTypes.string,
    type: PropTypes.oneOf(['text', 'image', 'icon']),
    size: PropTypes.oneOf([null, 'small', 'medium', 'large']),
    rounded: PropTypes.oneOf([null, 'small', 'medium', 'large', 'full']),
    fillMode: PropTypes.oneOf([null, 'solid', 'outline']),
    /* eslint-disable max-len */
    themeColor: PropTypes.oneOf([null, 'base', 'dark', 'error', 'info', 'inverse', 'inverse', 'light', 'primary', 'secondary', 'success', 'tertiary', 'warning'])
};
exports.Avatar.defaultProps = {
    type: Enums_1.avatarType.TEXT,
    size: 'medium',
    rounded: 'full',
    fillMode: 'solid',
    themeColor: 'primary'
};
