// @progress/kendo-react-layout/dist/npm/card/Card.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Card = void 0;
var React = require("react");
var PropTypes = require("prop-types");
var kendo_react_common_1 = require("@progress/kendo-react-common");
var Enums_1 = require("./interfaces/Enums");
var Enums_2 = require("./interfaces/Enums");
var kendo_react_common_2 = require("@progress/kendo-react-common");
var package_metadata_1 = require("../package-metadata");
exports.Card = React.forwardRef(function (props, ref) {
    var _a;
    (0, kendo_react_common_2.validatePackage)(package_metadata_1.packageMetadata);
    var children = props.children, dir = props.dir, style = props.style, className = props.className, type = props.type, orientation = props.orientation, customProps = __rest(props, ["children", "dir", "style", "className", "type", "orientation"]);
    var target = React.useRef(null);
    var chipRef = React.useRef(null);
    React.useImperativeHandle(target, function () { return ({
        element: chipRef.current,
        props: props
    }); });
    React.useImperativeHandle(ref, function () { return target.current; });
    return (React.createElement("div", __assign({ dir: dir, style: style, ref: chipRef, className: (0, kendo_react_common_1.classNames)('k-card', className, (_a = {}, _a["k-card-".concat(type)] = type !== Enums_1.cardType.DEFAULT, _a), orientation !== Enums_2.cardOrientation.HORIZONTAL ? 'k-card-vertical' : 'k-card-horizontal') }, customProps), children));
});
exports.Card.displayName = 'KendoReactCard';
exports.Card.propTypes = {
    children: PropTypes.node,
    className: PropTypes.string,
    dir: PropTypes.string,
    orientation: PropTypes.oneOf(['horizontal', 'vertical']),
    style: PropTypes.object,
    type: PropTypes.oneOf(['default', 'primary', 'info', 'success', 'warning', 'error'])
};
exports.Card.defaultProps = {
    type: Enums_1.cardType.DEFAULT,
    orientation: Enums_2.cardOrientation.VERTICAL
};
