// @progress/kendo-react-layout/dist/npm/menu/components/MenuItemArrow.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.MenuItemArrow = exports.leftArrowName = exports.rightArrowName = exports.downArrowName = void 0;
var React = require("react");
var PropTypes = require("prop-types");
var kendo_react_common_1 = require("@progress/kendo-react-common");
var misc_1 = require("../utils/misc");
/**
 * @hidden
 */
exports.downArrowName = 'caret-alt-down';
/**
 * @hidden
 */
exports.rightArrowName = 'caret-alt-right';
/**
 * @hidden
 */
exports.leftArrowName = 'caret-alt-left';
var MenuItemArrow = /** @class */ (function (_super) {
    __extends(MenuItemArrow, _super);
    function MenuItemArrow() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    /**
     * @hidden
     */
    MenuItemArrow.prototype.render = function () {
        return React.createElement(kendo_react_common_1.Icon, { className: 'k-menu-expand-arrow', name: this.getIconName() });
    };
    MenuItemArrow.prototype.getIconName = function () {
        var childrenPosition = (0, misc_1.getChildrenPosition)(this.props.itemId, this.props.verticalMenu === true, this.props.dir === 'rtl');
        switch (childrenPosition) {
            case 'downward':
                return exports.downArrowName;
            case 'rightward':
                return exports.rightArrowName;
            case 'leftward':
                return exports.leftArrowName;
            default:
                return undefined;
        }
    };
    /**
     * @hidden
     */
    MenuItemArrow.propTypes = {
        itemId: PropTypes.string,
        dir: PropTypes.string,
        verticalMenu: PropTypes.bool
    };
    return MenuItemArrow;
}(React.Component));
exports.MenuItemArrow = MenuItemArrow;
