// react-multistep/dist/index.js
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.js
var src_exports = {};
__export(src_exports, {
  default: () => src_default
});
module.exports = __toCommonJS(src_exports);

// src/MultiStep.jsx
var import_react = __toESM(require("react"));

// node_modules/goober/dist/goober.modern.js
var e = { data: "" };
var t = (t2) => "object" == typeof window ? ((t2 ? t2.querySelector("#_goober") : window._goober) || Object.assign((t2 || document.head).appendChild(document.createElement("style")), { innerHTML: " ", id: "_goober" })).firstChild : t2 || e;
var l = /(?:([\u0080-\uFFFF\w-%@]+) *:? *([^{;]+?);|([^;}{]*?) *{)|(}\s*)/g;
var a = /\/\*[^]*?\*\/|  +/g;
var n = /\n+/g;
var o = (e2, t2) => {
  let r = "", l2 = "", a2 = "";
  for (let n2 in e2) {
    let c2 = e2[n2];
    "@" == n2[0] ? "i" == n2[1] ? r = n2 + " " + c2 + ";" : l2 += "f" == n2[1] ? o(c2, n2) : n2 + "{" + o(c2, "k" == n2[1] ? "" : t2) + "}" : "object" == typeof c2 ? l2 += o(c2, t2 ? t2.replace(/([^,])+/g, (e3) => n2.replace(/(^:.*)|([^,])+/g, (t3) => /&/.test(t3) ? t3.replace(/&/g, e3) : e3 ? e3 + " " + t3 : t3)) : n2) : null != c2 && (n2 = /^--/.test(n2) ? n2 : n2.replace(/[A-Z]/g, "-$&").toLowerCase(), a2 += o.p ? o.p(n2, c2) : n2 + ":" + c2 + ";");
  }
  return r + (t2 && a2 ? t2 + "{" + a2 + "}" : a2) + l2;
};
var c = {};
var s = (e2) => {
  if ("object" == typeof e2) {
    let t2 = "";
    for (let r in e2)
      t2 += r + s(e2[r]);
    return t2;
  }
  return e2;
};
var i = (e2, t2, r, i2, p2) => {
  let u2 = s(e2), d2 = c[u2] || (c[u2] = ((e3) => {
    let t3 = 0, r2 = 11;
    for (; t3 < e3.length; )
      r2 = 101 * r2 + e3.charCodeAt(t3++) >>> 0;
    return "go" + r2;
  })(u2));
  if (!c[d2]) {
    let t3 = u2 !== e2 ? e2 : ((e3) => {
      let t4, r2, o2 = [{}];
      for (; t4 = l.exec(e3.replace(a, "")); )
        t4[4] ? o2.shift() : t4[3] ? (r2 = t4[3].replace(n, " ").trim(), o2.unshift(o2[0][r2] = o2[0][r2] || {})) : o2[0][t4[1]] = t4[2].replace(n, " ").trim();
      return o2[0];
    })(e2);
    c[d2] = o(p2 ? { ["@keyframes " + d2]: t3 } : t3, r ? "" : "." + d2);
  }
  let f2 = r && c.g ? c.g : null;
  return r && (c.g = c[d2]), ((e3, t3, r2, l2) => {
    l2 ? t3.data = t3.data.replace(l2, e3) : -1 === t3.data.indexOf(e3) && (t3.data = r2 ? e3 + t3.data : t3.data + e3);
  })(c[d2], t2, i2, f2), d2;
};
var p = (e2, t2, r) => e2.reduce((e3, l2, a2) => {
  let n2 = t2[a2];
  if (n2 && n2.call) {
    let e4 = n2(r), t3 = e4 && e4.props && e4.props.className || /^go/.test(e4) && e4;
    n2 = t3 ? "." + t3 : e4 && "object" == typeof e4 ? e4.props ? "" : o(e4, "") : false === e4 ? "" : e4;
  }
  return e3 + l2 + (null == n2 ? "" : n2);
}, "");
function u(e2) {
  let r = this || {}, l2 = e2.call ? e2(r.p) : e2;
  return i(l2.unshift ? l2.raw ? p(l2, [].slice.call(arguments, 1), r.p) : l2.reduce((e3, t2) => Object.assign(e3, t2 && t2.call ? t2(r.p) : t2), {}) : l2, t(r.target), r.g, r.o, r.k);
}
var d;
var f;
var g;
var b = u.bind({ g: 1 });
var h = u.bind({ k: 1 });
function m(e2, t2, r, l2) {
  o.p = t2, d = e2, f = r, g = l2;
}
function j(e2, t2) {
  let r = this || {};
  return function() {
    let l2 = arguments;
    function a2(n2, o2) {
      let c2 = Object.assign({}, n2), s2 = c2.className || a2.className;
      r.p = Object.assign({ theme: f && f() }, c2), r.o = / *go\d+/.test(s2), c2.className = u.apply(r, l2) + (s2 ? " " + s2 : ""), t2 && (c2.ref = o2);
      let i2 = e2;
      return e2[0] && (i2 = c2.as || e2, delete c2.as), g && i2[0] && g(c2), d(i2, c2);
    }
    return t2 ? t2(a2) : a2;
  };
}

// src/MultiStep.jsx
m(import_react.default.createElement);
var Ol = j("ol")`
  display: flex;
  margin: 0;
  padding-bottom: 2.2rem;
  list-style-type: none;
`;
var Li = j("li")`
  display: inline-block;
  text-align: center;
  line-height: 4.8rem;
  padding: 0 0.7rem;
  cursor: pointer;
  min-width: 6rem;

  color: silver;
  border-bottom: 2px solid silver;

 span{
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
 }

  &:hover,
  &:before {
    color: #0FA0CE;
  }
  &:after {
    content: "\\00a0\\00a0";
  }   
  &:before {
    position: relative;
    float: left;
    left: 50%;
    width: 1.3em;
    line-height: 1.4em;
    border-radius: 50%;
    bottom: -3.99rem;
  }
`;
var Todo = u`
  &:before {
    content: "\u039F";
    color: silver;
    background-color: white;
  }
`;
var Doing = u`
  &:before {
    content: "\u2022";
    color: white;
    background-color: #33C3F0;  
  }
`;
var Done = u`
  &:before {
    content: "\u2713";
    color: white;
    background-color: #33C3F0;
  }
`;
var RowDirection = u`
  flex-direction: row;
`;
var ColumnDirection = u`
  margin-top: 4.8rem;
  flex-direction: column;
`;
var getStep = (defaultIndex, newIndex, length) => {
  if (newIndex <= length) {
    return newIndex;
  }
  return defaultIndex;
};
var getTopNavStyles = (indx, length) => {
  const styles = [];
  for (let i2 = 0; i2 < length; i2++) {
    if (i2 < indx) {
      styles.push("done");
    } else if (i2 === indx) {
      styles.push("doing");
    } else {
      styles.push("todo");
    }
  }
  return styles;
};
var getButtonsState = (indx, length) => {
  if (indx > 0 && indx < length - 1) {
    return {
      showPreviousBtn: true,
      showNextBtn: true
    };
  } else if (indx === 0) {
    return {
      showPreviousBtn: false,
      showNextBtn: true
    };
  } else {
    return {
      showPreviousBtn: true,
      showNextBtn: false
    };
  }
};
function MultiStep(props) {
  const stepCustomStyle = typeof props.stepCustomStyle === "undefined" ? {} : props.stepCustomStyle;
  const showNav = typeof props.showNavigation === "undefined" ? true : props.showNavigation;
  const showTitles = typeof props.showTitles === "undefined" ? true : props.showTitles;
  const directionType = typeof props.direction === "undefined" ? "row" : props.direction;
  const [activeStep] = (0, import_react.useState)(getStep(0, props.activeStep, props.steps.length));
  const [stylesState, setStyles] = (0, import_react.useState)(getTopNavStyles(activeStep, props.steps.length));
  const [compState, setComp] = (0, import_react.useState)(activeStep);
  const [buttonsState, setButtons] = (0, import_react.useState)(getButtonsState(activeStep, props.steps.length));
  (0, import_react.useEffect)(() => {
    setStepState(props.activeStep);
  }, [props.activeStep]);
  const setStepState = (indx) => {
    setStyles(getTopNavStyles(indx, props.steps.length));
    setComp(indx < props.steps.length ? indx : compState);
    setButtons(getButtonsState(indx, props.steps.length));
  };
  const next = () => setStepState(compState + 1);
  const previous = () => setStepState(compState > 0 ? compState - 1 : compState);
  const handleOnClick = (evt) => {
    if (evt.currentTarget.value === props.steps.length - 1 && compState === props.steps.length - 1) {
      setStepState(props.steps.length);
    } else {
      setStepState(evt.currentTarget.value);
    }
  };
  const renderSteps = () => props.steps.map(
    (s2, i2) => {
      return /* @__PURE__ */ import_react.default.createElement(Li, {
        className: stylesState[i2] === "todo" ? Todo : stylesState[i2] === "doing" ? Doing : Done,
        style: { ...stepCustomStyle, transform: directionType == "column" ? "rotate(90deg)" : "rotate(0deg)" },
        onClick: handleOnClick,
        key: i2,
        value: i2
      }, showTitles && /* @__PURE__ */ import_react.default.createElement("span", null, s2.title ?? i2 + 1));
    }
  );
  const renderNav = (show) => show && /* @__PURE__ */ import_react.default.createElement("div", null, /* @__PURE__ */ import_react.default.createElement("button", {
    style: buttonsState.showPreviousBtn ? props.prevStyle : { display: "none" },
    onClick: previous
  }, "Prev"), /* @__PURE__ */ import_react.default.createElement("button", {
    style: buttonsState.showNextBtn ? props.nextStyle : { display: "none" },
    onClick: next
  }, "Next"));
  return /* @__PURE__ */ import_react.default.createElement("div", {
    style: { display: "flex", flexDirection: directionType === "column" ? "row" : "column" }
  }, /* @__PURE__ */ import_react.default.createElement(Ol, {
    className: directionType === "column" ? ColumnDirection : RowDirection
  }, renderSteps()), props.steps[compState].component, /* @__PURE__ */ import_react.default.createElement("div", null, renderNav(showNav)));
}

// src/index.js
var src_default = MultiStep;
