// @progress/kendo-react-layout/dist/npm/tabstrip/TabStripTab.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.TabStripTab = void 0;
var React = require("react");
var PropTypes = require("prop-types");
var TabStripTab = /** @class */ (function (_super) {
    __extends(TabStripTab, _super);
    function TabStripTab() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    /**
     * @hidden
     */
    TabStripTab.prototype.render = function () {
        return null;
    };
    /**
     * @hidden
     */
    TabStripTab.propTypes = {
        disabled: PropTypes.bool,
        contentClassName: PropTypes.string,
        children: PropTypes.oneOfType([
            PropTypes.element,
            PropTypes.node
        ]),
        title: PropTypes.oneOfType([
            PropTypes.string,
            PropTypes.element,
            PropTypes.node
        ])
    };
    return TabStripTab;
}(React.Component));
exports.TabStripTab = TabStripTab;
