// @progress/kendo-react-layout/dist/npm/expansionpanel/ExpansionPanel.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ExpansionPanel = void 0;
var React = require("react");
var PropTypes = require("prop-types");
var kendo_react_common_1 = require("@progress/kendo-react-common");
var package_metadata_1 = require("../package-metadata");
/**
 * Represents the [KendoReact ExpansionPanel component]({% slug overview_expansionpanel %}).
 */
exports.ExpansionPanel = React.forwardRef(function (props, target) {
    var _a, _b;
    (0, kendo_react_common_1.validatePackage)(package_metadata_1.packageMetadata);
    var elementRef = React.useRef(null);
    var _c = React.useState(false), focused = _c[0], setFocused = _c[1];
    var getImperativeHandle = React.useCallback(function () { return ({ element: elementRef.current }); }, []);
    React.useImperativeHandle(target, getImperativeHandle);
    var _d = props.expanded, expanded = _d === void 0 ? false : _d, disabled = props.disabled, title = props.title, subtitle = props.subtitle, onAction = props.onAction, expandIcon = props.expandIcon, collapseIcon = props.collapseIcon;
    var onClick = React.useCallback(function (event) {
        (0, kendo_react_common_1.dispatchEvent)(onAction, event, getImperativeHandle(), { expanded: expanded });
    }, [onAction, expanded]);
    var onKeyDown = React.useCallback(function (event) {
        if (event.keyCode === kendo_react_common_1.Keys.enter || event.keyCode === kendo_react_common_1.Keys.space) {
            event.preventDefault();
            (0, kendo_react_common_1.dispatchEvent)(onAction, event, getImperativeHandle(), { expanded: expanded });
        }
    }, [onAction, expanded]);
    var handleFocus = React.useCallback(function () {
        if (props.disabled) {
            return;
        }
        setFocused(true);
    }, [props.disabled]);
    var handleBlur = React.useCallback(function () {
        if (props.disabled) {
            return;
        }
        setFocused(false);
    }, [props.disabled]);
    var _e = (0, kendo_react_common_1.useAsyncFocusBlur)({ onFocus: handleFocus, onBlur: handleBlur }), onFocus = _e.onFocus, onBlur = _e.onBlur;
    return (React.createElement("div", { ref: elementRef, className: (0, kendo_react_common_1.classNames)('k-expander', props.className, {
            'k-expanded': expanded,
            'k-focus': focused && !disabled,
            'k-disabled': disabled
        }), onFocus: onFocus, onBlur: onBlur, style: props.style, id: props.id, dir: (0, kendo_react_common_1.useRtl)(elementRef, props.dir), onKeyDown: disabled ? undefined : onKeyDown },
        React.createElement("div", { role: 'button', "aria-controls": props.ariaControls, "aria-expanded": expanded, "aria-disabled": disabled, tabIndex: (0, kendo_react_common_1.getTabIndex)(props.tabIndex, disabled), className: "k-expander-header", onClick: disabled ? undefined : onClick },
            title !== undefined ? React.createElement("div", { className: "k-expander-title" }, title) : undefined,
            React.createElement("span", { className: "k-spacer" }),
            subtitle !== undefined ? React.createElement("div", { className: "k-expander-sub-title" }, subtitle) : undefined,
            React.createElement("span", { className: "k-expander-indicator" },
                React.createElement(kendo_react_common_1.Icon, { name: !expanded
                        ? !expandIcon ? 'chevron-down' : undefined
                        : !collapseIcon ? 'chevron-up' : undefined, className: (0, kendo_react_common_1.classNames)(expanded
                        ? (_a = {},
                            _a[String(collapseIcon)] = Boolean(collapseIcon),
                            _a) : (_b = {},
                        _b[String(expandIcon)] = Boolean(expandIcon),
                        _b)) }))),
        props.children));
});
exports.ExpansionPanel.propTypes = {
    children: PropTypes.node,
    className: PropTypes.string,
    style: PropTypes.object,
    dir: PropTypes.string,
    id: PropTypes.string,
    tabIndex: PropTypes.number,
    title: PropTypes.node,
    subtitle: PropTypes.node,
    expandIcon: PropTypes.string,
    collapseIcon: PropTypes.string,
    expanded: PropTypes.bool,
    disabled: PropTypes.bool,
    onAction: PropTypes.func
};
exports.ExpansionPanel.displayName = 'KendoReactExpansionPanel';
