// @progress/kendo-react-layout/dist/npm/breadcrumb/BreadcrumbDelimiter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BreadcrumbDelimiter = void 0;
var React = require("react");
var PropTypes = require("prop-types");
var kendo_licensing_1 = require("@progress/kendo-licensing");
var kendo_react_common_1 = require("@progress/kendo-react-common");
var package_metadata_1 = require("../package-metadata");
/**
 * Represents the BreadcrumbDelimiter component.
 */
exports.BreadcrumbDelimiter = React.forwardRef(function (props, ref) {
    (0, kendo_licensing_1.validatePackage)(package_metadata_1.packageMetadata);
    var target = React.useRef(null);
    var breadcrumbDelimiterRef = React.useRef(null);
    React.useImperativeHandle(target, function () {
        var _a;
        return ({
            element: ((_a = breadcrumbDelimiterRef.current) === null || _a === void 0 ? void 0 : _a.element) || null,
            props: props
        });
    });
    React.useImperativeHandle(ref, function () { return target.current; });
    return (React.createElement(kendo_react_common_1.Icon, { ref: breadcrumbDelimiterRef, id: props.id, tabIndex: props.tabIndex, style: props.style, name: "chevron-right", className: "k-breadcrumb-delimiter-icon" }));
});
var propTypes = {
    id: PropTypes.string,
    className: PropTypes.string,
    style: PropTypes.object,
    tabIndex: PropTypes.number
};
exports.BreadcrumbDelimiter.displayName = 'KendoReactBreadcrumbDelimiter';
exports.BreadcrumbDelimiter.propTypes = propTypes;
